### PR TITLE
Add hh filetype to the plugin.

### PR DESCRIPTION
**Pull request by jgimeno**
To help configure other tools as phpcd.vim we need a way to avoid php collision. Using hh as a new filetype has more sense since it's really not php but hack what we want to use.

### CODE REVIEW

**Comment by jwatzman** [comment]
This looks to remove the `.php` extension support. Keeping this is important -- a lot of Hack code exists with a `.php` extension. (Notably *all* of FB's code is this way, but I've seen it a lot externally too.)

**Comment by jgimeno** [comment]
Hi josh! 

Thaks for answer me, just I pushed something that solves maybe this issue. It is useful for me because in my company we use too lot of code with php extension but real <?hh.

Just feel free what is best for the project, check my last commit.

Thanks! :D

**Comment by jwatzman** [comment]
@jgimeno thanks for the update, appreciate the PR. Your change is supposed to gate the behaviour on the file actually starting with `<?hh`, is that correct? I'm far from an expert on vimscript :)

@mxw are you still around, can you take a look at this?

**Comment by jgimeno** [comment]
Yes, that's correct @jwatzman. For me it allows to use different omnifunc engines. 

The thing about setting on every .hh, .hhi or even .php as filetype=php didn't allow me to set a different omnifunc for hack files (hackcomplete) and for php (phpcd.vim). This works for me but I don't know if it's enough for all people using the plugin.

Thanks!

File: ftdetect/hack.vim [hunk 1]
```diff
@@ -11,9 +11,16 @@
 " grant of patent rights can be found in the PATENTS file in the same
 " directory.
 
-au BufRead,BufNewFile *.hhi setl filetype=php
+au BufRead,BufNewFile *.hhi setl filetype=hh | set syntax=php
```
Comment (usox) [inline ftdetect/hack.vim]: I think it should be `setl syntax` instead of `set syntax` in here?

**Comment by usox** [comment]
Ok, I'm a little late to the party ;)
Having an own filetype for `hh` is a good idea. My only question regarding this change is: will changing the filetype cause problems with other plugins which depend on the `php` filetype? Actually I don't know many hack-plugins, but...

**Comment by jgimeno** [comment]
Hi @usox ! Don't worry! Being using those changes half a year still did not have any issue, but as you said it could have some problems that maybe are not foreseen. So let's wait for some opinions! But I guess there isn't a lot of movement lately around here.

Thanks!

**Comment by fredemmott** [comment]
Resolved by #40 - setting filetype to 'hack', and syntax to 'php'

**Pull request closed without merging**